### ISSUE TITLE
No easy way to set up an execution environment

### ISSUE THREAD

**Issue by fabianegli** [fabianegli/singlecell_proteomics#1]
It would be nice to have one - or multiple - easy ways to set up a Python environment that can run the code.

**Comment by fabianegli** [comment]
This is a documentation of the odyssey to reproduce the environment necessary to reproduce the Jupyter notebook in the original repository:


1. Try with a venv on my local machine to make a first attempt to run the jupyter notebook.
   I am trying to work with Python 3.10 - not the newest, but better than the previous ones and I assume all dependencies will support it.

```
python3.10 -m venv venv
source venv/bin/activate
pip install -U pip
```

(Luckily we already have a .gitignore that is made for Python development so we can create the venv in the repository's root straight away.) 5. Get JupyterLab

```
pip install jupyterlab
```

2. Revising decision to use Python 3.10 to use 3.8.2 as visible form the notebook.
   In the same block it reports scanpy version 1.8.2
   These are all the reported Version I can find in the notebook. So lets use them.

> NB: They should have done a `pip freeze` or `conda export` at least to record the used versions and indicated the OS. Other options like using the `watermark` package would also have helped. Of course even better would be a containerization approach of some sorts. But we'll get there later???

3. I don't have Python 3.8.2 installed on my system, so for the sake of reproducibility let's use conda for the moment.

For this purpose lets create a env yaml file for conda

```
% conda create --name scprep python=3.8.2
% conda activate scprep
(scprep) % python -V
Python 3.8.2
```

`reproduce-single-cell-proteomics.yaml`:

```
name: scprep
channels:
  - conda-forge
  - defaults
dependencies:
  - anndata
  - jupyterlab
  - matplotlib
  - numpy
  - pandas
  - python=3.8.2
  - scanpy=1.8.2
  - scipy
  - seaborn
  - scikit-learn
```

```
conda env create --file reproduce-single-cell-proteomics.yaml
conda activate scprep
```

Then start JupyterLab
`% jupyter lab`

Open the notebook.

Running the imports raises a SystemError.

> SystemError: initialization of \_internal failed without raising an exception

According to https://stackoverflow.com/a/74982146/6018688 it might be a numpy issue.
The proposed solution is to downgrade numpy to 1.23.5 (currently installed is 1.24.1).

```
conda install numpy=1.23.5
```

Retry the notebook import cell.

Now the import works, so the question and the answer on StackOverflow get an upvote.

But there's this warning:

> MatplotlibDeprecationWarning: The seaborn styles shipped by Matplotlib are deprecated since 3.6, as they no longer correspond to the styles shipped by seaborn. However, they will remain available as 'seaborn-v0_8-<style>'. Alternatively, directly use the seaborn API instead.

So downgrade to matplotlib 3.5 from 3.6.2

`reproduce-single-cell-proteomics.yaml`:

```
name: scprep
channels:
  - defaults
  - conda-forge
dependencies:
  - anndata
  - jupyterlab
  - matplotlib=3.5.2
  - numpy=1.23.5
  - pandas
  - python=3.8.2
  - scanpy=1.8.2
  - scipy
  - seaborn
  - scikit-learn
```

At this point all imports work.


Reading data Cell 4: Warning mismatch.

From

>/Users/sabrina.richter/anaconda3/envs/tissue/lib/python3.8/site-packages/anndata/_core/anndata.py:119: ImplicitModificationWarning: Transforming to str index.
>  warnings.warn("Transforming to str index.", ImplicitModificationWarning)

to
>/var/folders/lh/bk80lqb10yn5y40bxg593pv40000gn/T/ipykernel\_48740/2330115773.py:11: FutureWarning: X.dtype being converted to np.float32 from float64. In the next version of anndata (0.9) conversion will not be automatic. Pass dtype explicitly to avoid this warning. Pass `AnnData(X, dtype=X.dtype, ...)` to get the future behavour.
>  adata\_prot\_raw = anndata.AnnData(prot.transpose())
>/Users/eglif/miniconda3/envs/scprep/lib/python3.8/site-packages/anndata/_core/anndata.py:121: ImplicitModificationWarning: Transforming to str index.
>  warnings.warn("Transforming to str index.", ImplicitModificationWarning)

So probably the used version of anndata was pre 0.8.


```
name: scprep
channels:
  - defaults
  - conda-forge
dependencies:
  - anndata=0.7.8
  - jupyterlab
  - matplotlib=3.5.2
  - numpy=1.23.5
  - pandas
  - python=3.8.2
  - scanpy=1.8.2
  - scipy
  - seaborn
  - scikit-learn
```

conda deactivate
conda env remove -n scprep
conda env create --file reproduce-single-cell-proteomics.yaml
conda activate scprep


Next round: From version 0.12.0 on wards seaborn enforces the use of keyword arguments.


Additional Warning in cell 16 from:
```
/Users/eglif/miniconda3/envs/scprep/lib/python3.8/site-packages/scanpy/tools/_score_genes.py:151: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.
  for cut in np.unique(obs_cut.loc[gene_list]):
```

Ignoring it, because we have the required scanpy version as documented in the beginning of the notebook.

Misssing FutureWarning in cell 19.

**Comment by fabianegli** [comment]
A conda environment to reproduce the notebook has been added in c6e322b479ef3637a4ad89c7fa47032ccfa68ad7.
